test/numerical/TestConv.cpp

```diff
@@ -87,11 +87,83 @@ int main(int argc, char *argv[]) {
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 64, 16, 55, 55, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 16, 64, 55, 55, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 128, 16, 55, 55, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 16, 64, 55, 55, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 128, 32, 27, 27, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 32, 128, 27, 27, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 256, 32, 27, 27, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 32, 128, 27, 27, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 256, 48, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 48, 192, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 384, 48, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 48, 192, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 384, 64, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 64, 256, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 512, 64, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 64, 256, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    assert(isOMConvTheSameAsNaiveImplFor(
+               1, 512, 1000, 13, 13, 1, 1, 0, 0, 0, 0, ConvAutoPad::NOTSET) &&
+           "failed test from test_squeezenet_cpu");
+    return 0;
+  }
```

Comment:
This is cool. You can query all patterns in the backend benchmarks. Is `if (1)` redundant?

Reply:
Its call grep and copy, but I was able to diff debug output to guarantee that it TestConv called the same kernels as in the benchmark. The only difference with the benchmark is that it also use a bias, and our testing does not currently support tests of bias... maybe something to add.